src/main/clojure/clara/rules/engine.cljc

```diff
@@ -687,58 +685,75 @@
-
-         ;; There are no previously accumulated results, but we still may need to propagate things
-         ;; such as a sum of zero items.
-         ;; If all variables in the accumulated item are bound and an initial
-         ;; value is provided, we can propagate the initial value as the accumulated item.
-
-         (and (has-keys? (:bindings token)
-                         binding-keys) ; All bindings are in place.
-              (:initial-value accumulator)) ; An initial value exists that we can propagate.
-         (let [fact-bindings (select-keys (:bindings token) binding-keys)
-               previous (:initial-value accumulator)]
-
-           ;; Send the created accumulated item to the children.
-           (send-accumulated node accum-condition accumulator result-binding token previous fact-bindings transport memory listener)
-
-           (l/add-accum-reduced! listener node join-bindings previous fact-bindings)
-
-           ;; Add it to the working memory.
-           (mem/add-accum-reduced! memory node join-bindings previous fact-bindings))
-
-         ;; Propagate nothing if the above conditions don't apply.
-         :default nil))))
+          ;; If there are previously accumulated results to propagate, simply use them.
+          (seq previous-results)
+          (doseq [[fact-bindings previous] previous-results
+                  :let [converted (when previous
+                                    (convert-return-fn previous))]
+                  :when converted]
+            (send-accumulated node accum-condition accumulator result-binding token
+                              converted fact-bindings transport memory listener))
+
+          ;; There are no previously accumulated results, but we still may need to propagate things
+          ;; such as a sum of zero items.
+          ;; If all variables in the accumulated item are bound and an initial
+          ;; value is provided, we can propagate the initial value as the accumulated item.
+
+          (and (has-keys? (:bindings token)
+                          binding-keys) ; All bindings are in place.
+               (:initial-value accumulator)) ; An initial value exists that we can propagate.
+          (let [fact-bindings (select-keys (:bindings token) binding-keys)
+                initial-value (:initial-value accumulator)
+                ;; Note that we check that the :initial-value is truthy above, which is why we
+                ;; don't need (when initial-value (convert-return-fn initial-value)) here.
+                initial-converted (convert-return-fn initial-value)]
+            
+            (l/add-accum-reduced! listener node join-bindings initial-value fact-bindings)
+
+            ;; Add it to the working memory.
+            (mem/add-accum-reduced! memory node join-bindings initial-value fact-bindings)
+
+            (when initial-converted
```

Comment:
I can understand that some cases propagating false may be "expected" to work.  I think this would be a larger deviation from the current behavior though.  I know in our use-cases of accumulators we have always used false/nil interchangeably as "falsey" and not expected to propagate.  So if false was propagated, I think it'd be something that had to be explained to others to avoid using.

I'm not entirely opposed, I just wonder how broad of an impact that would be to change it.  I also, think it is hard to understand what people would expect.  I think it can be surprising either way depending on what the goal is at the time.